### PR TITLE
Stat all folders and files within appRoot as readonly (#138815)

### PR DESCRIPTION
**Pull request by gjsjohnmurray**
Second attempt at resolving #138815 to reduce the likelihood of a naive user storing their projects within the appRoot folder and then being upset when a VS Code update deletes their files.

This PR implements @bpasero's suggestion at https://github.com/microsoft/vscode/pull/155443#issuecomment-1191411763 but also makes Explorer play nice.

junk

### CODE REVIEW

File: src/vs/workbench/contrib/files/common/explorerModel.ts [hunk 1]
```diff
@@ -242,6 +242,7 @@ export class ExplorerItem {
 			local.updateName(disk.name);
 		}
 		local._isDirectory = disk.isDirectory;
+		local._readonly = disk._readonly;
```
Comment (gjsjohnmurray) [inline src/vs/workbench/contrib/files/common/explorerModel.ts:245]: In the end this change wasn't necessary to make this PR work, but it looked like an omission so I have left it in. @bpasero what do you think?

**Comment by gjsjohnmurray** [comment]
Submitting the PR has revealed that I've broken layering rules in the way I made Explorer work out whether to force metadata resolution so it gets to know when folders and files are read-only.

@bpasero please advise whether there's another way to achieve this.

**Comment by gjsjohnmurray** [comment]
To avoid the layer breaker I have made Explorer get metadata for all file:// uris.

**Comment by bpasero** [comment]
Thanks, will review when I find time for it.

**Comment by gjsjohnmurray** [comment]
Added a warning notification when workspace opens, similar to the one the previous PR had per-file:

image

**Comment by bpasero** [comment]
Sorry, will probably not have a lot of time this month, might push this back a bit.

**Comment by gjsjohnmurray** [comment]
@bpasero @lramos15 there have been two new issues raised this week by Windows users who have lost all their files. Neither yet proven to be caused by an update, but please can this PR make progress as a way of preventing future incidents of that sort?

**Comment by bpasero** [comment]
@gjsjohnmurray I am not easy accepting this change, especially forcing `requireMetadata` will have a severe significant impact on explorer performance and this case here only impacts a handful of users.

How about a different approach where we simply open a notification if a user opens the installation dir or any child of that as a workspace folder? We could even leverage the banner for this to make it more prominent:

image

Maybe it could also be a modal dialog to be very annoying.

**Comment by gjsjohnmurray** [comment]
@bpasero your verdict doesn't surprise me. I was uneasy about having to get metadata so unconditionally. Closing this. Please see #161534 instead.

**Pull request closed without merging**